src/drivers/sifive_pwm0.c

```diff
@@ -0,0 +1,340 @@
+/* Copyright 2019 SiFive, Inc */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#include <metal/machine/platform.h>
+
+#ifdef METAL_SIFIVE_PWM0
+#include <metal/clock.h>
+#include <metal/compiler.h>
+#include <metal/drivers/sifive_gpio0.h>
+#include <metal/drivers/sifive_pwm0.h>
+#include <metal/io.h>
+#include <metal/machine.h>
+#include <metal/time.h>
+#include <stdio.h>
+
+/* Register fields */
+#define METAL_PWMCFG_STICKY (1UL << 8)
+#define METAL_PWMCFG_ZEROCMP (1UL << 9)
+#define METAL_PWMCFG_DEGLITCH (1UL << 10)
+#define METAL_PWMCFG_ENALWAYS (1UL << 12)
+#define METAL_PWMCFG_ENONESHOT (1UL << 13)
+#define METAL_PWMCFG_CMPCENTER(x) (1UL << (16 + x))
+#define METAL_PWMCFG_CMPIP(x) (1UL << (28 + x))
+#define METAL_SIFIVE_PWM0_PWMCMP(x) METAL_SIFIVE_PWM0_PWMCMP0 + x * 4
+
+/* Macros to access registers */
+#define METAL_PWM_REG(offset) ((base + offset))
+#define METAL_PWM_REGW(offset)                                                 \
+    (__METAL_ACCESS_ONCE((__metal_io_u32 *)METAL_PWM_REG(offset)))
+
+/* Macro to get PWM compare count */
+#define METAL_PWM_GETCMPVAL(duty) (duty * pwm->count_val) / 100U
```

Comment:
Best to put () around the macro variable, like METAL_PWMCFG_CMPCENTER(x) (1UL << (16 + (x)))

Reply:
fixed.